lib/liquid/extensions.rb

```diff
@@ -25,6 +26,12 @@ def to_liquid
   end
 end
 
+class BigDecimal
+  def to_liquid
+    to_f
```

Comment:
Well, if you chain together filters, then we call .to_liquid on the result to avoid using non-liquid values.  The same is true if you assign the result to a variable then retrieve it from a variable tag.  That makes a BigDecimal seem even more pointless.